### ISSUE TITLE
Finetuning on custom dataset

### ISSUE THREAD

**Issue by mileseverett** [JunkyByte/easy_ViTPose#15]
How would I go about finetuning from one of the coco pretrained models on my own dataset?

Thanks in advance

**Comment by JunkyByte** [comment]
Hello, this project aims to provide pretrained models, finetuning might require some additonal work.  
That being said, here's what you have to do:

- Create the dataset class to iterate over your data, you can mimick the COCO class: [easy_ViTPose/datasets/COCO.py](https://github.com/JunkyByte/easy_ViTPose/blob/main/easy_ViTPose/datasets/COCO.py)  
I would recommend to adapt your labels to COCO format if possible and adapt this script to use your annotations and images instead of creating everything from scratch.

- Configure the [config.yaml](https://github.com/JunkyByte/easy_ViTPose/blob/main/easy_ViTPose/config.yaml) script

- Set the correct keypoint number [easy_ViTPose/configs/ViTPose_common.py#L28](https://github.com/JunkyByte/easy_ViTPose/blob/2c0a0c8017255e5f6bd1b8ccbfd917845a0b989f/easy_ViTPose/configs/ViTPose_common.py#L28)

- Set the data root to point to your data [easy_ViTPose/configs/ViTPose_common.py#L53](https://github.com/JunkyByte/easy_ViTPose/blob/2c0a0c8017255e5f6bd1b8ccbfd917845a0b989f/easy_ViTPose/configs/ViTPose_common.py#L53)

- Download a checkpoint pretrained on COCO from this repo or the original implementation, set the `resume_from` path in `config.yaml` to it.

- Finetune using the `train.py` script. It will load the model, pop the head layers of the output and finetune new ones. All the other stages of the network will be freezed. This is what I did to finetune on the COCO+Feet dataset and seems to work fine, check [easy_ViTPose/train.py#L111](https://github.com/JunkyByte/easy_ViTPose/blob/2c0a0c8017255e5f6bd1b8ccbfd917845a0b989f/easy_ViTPose/train.py#L111)  
Note that if you created a new Class to iterate over your data you have to point to it in the `train.py` script as well.

**Comment by mileseverett** [comment]
Much appreciated. Will give this a go and reopen if any issues.

**Comment by Galiold** [comment]
The directions about using `train.py` for fine tuning don't seem to work on the current branch, how can I use the current changes to train on new data?

Thanks
